compositions/upbound-aws-provider/serverless-microservice/rest-lambda-ddb.yaml

```diff
@@ -0,0 +1,810 @@
+      base:
+        apiVersion: awsblueprints.io/v1alpha1
+        kind: XDynamoDBTable
+        spec:
+          deletionPolicy: Delete
+          compositionSelector:
+            matchLabels:
+              awsblueprints.io/provider: aws
+              awsblueprints.io/environment: dev
+          dynamoConfig:
+            attribute:
+              - name: id
+                type: S
+            hashKey: id
+      patches:
+        - type: PatchSet
+          patchSetName: common-fields
+        - type: FromCompositeFieldPath
+          fromFieldPath: metadata.name
+          toFieldPath: metadata.name
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-data"
+        - type: ToCompositeFieldPath
+          fromFieldPath: status.tableName
+          toFieldPath: status.tableName
+        - type: ToCompositeFieldPath
+          fromFieldPath: status.tableArn
+          toFieldPath: status.tableArn
+
+    - name: logic-lambda-ddb-access-policy
+      base:
+        apiVersion: awsblueprints.io/v1alpha1
+        kind: IAMPolicy
+        metadata:
+          labels:
+            type: lambda-ddb-access-policy
+        spec:
+          deletionPolicy: Delete
+          compositionSelector:
+            matchLabels:
+              awsblueprints.io/provider: aws
+              awsblueprints.io/environment: dev
+              iam.awsblueprints.io/policy-type: write
+              iam.awsblueprints.io/service: dynamodb
+      patches:
+        - type: PatchSet
+          patchSetName: common-fields
+        - type: FromCompositeFieldPath
+          fromFieldPath: metadata.name
+          toFieldPath: metadata.name
+        - type: FromCompositeFieldPath
+          fromFieldPath: status.businessLogicLambdaRoleName
+          toFieldPath: spec.roleName
+          policy:
+            fromFieldPath: Required
+        - type: FromCompositeFieldPath
+          fromFieldPath: status.tableArn
+          toFieldPath: spec.resourceArn
+          policy:
+            fromFieldPath: Required
+
+    - name: authorizer-lambda
+      base:
+        apiVersion: awsblueprints.io/v1alpha1
+        kind: XLambdaFunction
+        metadata:
+          labels:
+            type: apigw-authorizer-lambda
+        spec:
+          deletionPolicy: Delete
+          compositionSelector:
+            matchLabels:
+              awsblueprints.io/provider: aws
+              awsblueprints.io/environment: dev
+              awsblueprints.io/type: zip
+      patches:
+        - type: PatchSet
+          patchSetName: common-fields
+        - type: FromCompositeFieldPath
+          fromFieldPath: metadata.name
+          toFieldPath: metadata.name
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-authorizer"
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.authorizerPassword
+          toFieldPath: spec.envVariables[AUTHORIZER_PASSWORD]
```

Comment:
I do not want to encourage this practice here.

Reply:
Updated to use AWS Secrets Manager secure string.